src/main/python/ayab/thread.py

```diff
@@ -17,21 +17,23 @@
 #    Copyright 2014 Sebastian Oliva, Christian Obersteiner, Andreas Müller, Christian Gerbrandt
 #    https://github.com/AllYarnsAreBeautiful/ayab-desktop
 
+from __future__ import annotations
 from PySide6.QtCore import QThread
+from typing import Callable,Any
 
 
 class GenericThread(QThread):
     '''A generic thread wrapper for functions on threads.'''
-    def __init__(self, function, *args, **kwargs):
+    def __init__(self, function:Callable[...,Any], *args:Any, **kwargs:Any):
```

Comment:
this could _probably_ be given a Generic typing, and that might be a more accurate type- but Generics aren't as good here as in Typescript, so it didn't seem worth the effort.